website/docs/class-of.md

```diff
@@ -1,75 +1,200 @@
 ---
 id: class-of
-title: T.class_of
+title: Types for Class Objects via T.class_of
+sidebar_label: T.class_of
 ---
 
-> TODO: This page is still a fragment. Contributions welcome!
+Classes are also values in Ruby. Sorbet uses `T.class_of(...)` to describe the
+type of those class objects.
 
 ```ruby
 T.class_of(Integer)
 ```
 
-It can be confusing whether you want `MyClass` or `T.class_of(MyClass)`. For
-reference, these assertions are true:
+The difference between `MyClass` and `T.class_of(MyClass)` can be confusing.
+Here are some examples to make it less confusing:
+
+| These expressions... | ...have these types   |
+| -------------------- | --------------------- |
+| `0`, `1`, `2 + 2`    | `Integer`             |
+| `Integer`            | `T.class_of(Integer)` |
+| `42.class`           | `T.class_of(Integer)` |
+
+Here's a playground link to confirm these types:
 
 ```ruby
-T.let(5, Integer)
+# typed: true
+T.let(0, Integer)
+T.let(1, Integer)
+T.let(2 + 2, Integer)
 
-T.let(5.class, T.class_of(Integer))
 T.let(Integer, T.class_of(Integer))
+T.let(42.class, T.class_of(Integer))
 ```
 
-You can also use modules. It respects inheritance in the same way that
-[Class Types](class-types.md) do:
+<a href="https://sorbet.run/#%23%20typed%3A%20true%0AT.let(0%2C%20Integer)%0AT.let(1%2C%20Integer)%0AT.let(2%20%2B%202%2C%20Integer)%0A%0AT.let(Integer%2C%20T.class_of(Integer))%0AT.let(42.class%2C%20T.class_of(Integer))">
+  → View on sorbet.run
+</a>
+
+## `T.class_of` and inheritance
+
+As with [Class Types](class-types.md#inheritance), `T.class_of` types work with
+inheritance:
 
 ```ruby
+# typed: true
 extend T::Sig
 
 class Grandparent; end
 class Parent < Grandparent; end
 class Child < Parent; end
 
 sig {params(x: T.class_of(Parent)).void}
-def foo(x); end
-
-foo(Parent) # ok
-foo(Child) # ok
+def example(x); end
 
-foo(Grandparent) # error
+example(Grandparent)   # error
+example(Parent)        # ok
+example(Child)         # ok
 ```
 
-<a href="https://sorbet.run/#extend%20T%3A%3ASig%0A%0Aclass%20Grandparent%3B%20end%0Aclass%20Parent%20%3C%20Grandparent%3B%20end%0Aclass%20Child%20%3C%20Parent%3B%20end%0A%0Asig%20%7Bparams(x%3A%20T.class_of(Parent)).void%7D%0Adef%20foo(x)%3B%20end%0A%0A%0Afoo(Parent)%20%23%20ok%0Afoo(Child)%20%23%20ok%0A%0Afoo(Grandparent)%20%23%20error">
+<a href="https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Aclass%20Grandparent%3B%20end%0Aclass%20Parent%20%3C%20Grandparent%3B%20end%0Aclass%20Child%20%3C%20Parent%3B%20end%0A%0Asig%20%7Bparams(x%3A%20T.class_of(Parent)).void%7D%0Adef%20example(x)%3B%20end%0A%0Aexample(Grandparent)%20%20%20%23%20error%0Aexample(Parent)%20%20%20%20%20%20%20%20%23%20ok%0Aexample(Child)%20%20%20%20%20%20%20%20%20%23%20ok">
+  → View on sorbet.run
+</a>
+
+The most surprising feature of `T.class_of` comes from not understanding
+inheritance in Ruby, especially with `include` or `extend` plus modules.
```

Comment:
- `include` and `extend` are not keywords, they're methods.
- I'd prefer not using `/` in written documents like this, because it's ambiguous. `or` is more clear
- We don't use `you` in Sorbet docs, because it leads to a tone like, "you who make mistakes, vs me who is always right" throughout the documentation.

Going to pass on this. Feel free to suggest something else.